packages/eslint-plugin/src/rules/no-magic-numbers.ts

```diff
@@ -20,6 +20,7 @@ export default util.createRule<Options, MessageIds>({
       description: 'Disallow magic numbers',
       category: 'Best Practices',
       recommended: false,
+      extendsBaseRule: true,
```

Comment:
small question about this, do we want to support string here?

Reply:
that's a good point.
We already have this case with `no-return-await` (implemented as `return-await`)

Reply:
maybe we should go one step further and replace it with string[] as naming-convention rule replaces a lot of eslint rules.

Reply:
`naming-convention` is a weird case... It's not an extension rule, it's a brand new rule which overlaps with some base rules.

I don't think we want to list this "overlap" in the docs anywhere.

----

As a side note, I was actually considering submitting `naming-convention` as a rule for eslint core.
The only typescript-specific thing in there is the types option, which is easy enough to trim out.